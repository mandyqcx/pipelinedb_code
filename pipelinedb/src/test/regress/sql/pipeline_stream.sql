CREATE FOREIGN TABLE stream0 (id integer) SERVER pipelinedb;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
CREATE VIEW ps0 AS SELECT id FROM stream0;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
CREATE VIEW ps1 AS SELECT count(*) FROM stream0;
CREATE VIEW ps2 AS SELECT id FROM stream0;
CREATE FOREIGN TABLE stream1 (x integer, y timestamp) SERVER pipelinedb;
CREATE VIEW ps3 AS SELECT x, y FROM stream1;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
CREATE VIEW ps4 AS SELECT id::text FROM stream0;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
CREATE FOREIGN TABLE stream2 (x integer) SERVER pipelinedb;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
CREATE VIEW ps5 AS SELECT x FROM stream2;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
DROP VIEW ps0;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
DROP VIEW ps1;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
DROP VIEW ps2;
DROP VIEW ps3;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
DROP VIEW ps5;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
DROP FOREIGN TABLE stream2;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
DROP FOREIGN TABLE stream1;
DROP FOREIGN TABLE stream0 CASCADE;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
