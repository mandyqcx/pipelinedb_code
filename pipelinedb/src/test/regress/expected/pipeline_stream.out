CREATE FOREIGN TABLE stream0 (id integer) SERVER pipelinedb;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |  name   | queries 
--------+---------+---------
 public | stream0 | 
(1 row)

CREATE VIEW ps0 AS SELECT id FROM stream0;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |   name    | queries 
--------+-----------+---------
 public | ps0_osrel | 
 public | stream0   | {ps0}
(2 rows)

CREATE VIEW ps1 AS SELECT count(*) FROM stream0;
CREATE VIEW ps2 AS SELECT id FROM stream0;
CREATE FOREIGN TABLE stream1 (x integer, y timestamp) SERVER pipelinedb;
CREATE VIEW ps3 AS SELECT x, y FROM stream1;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |   name    |    queries    
--------+-----------+---------------
 public | ps0_osrel | 
 public | ps1_osrel | 
 public | ps2_osrel | 
 public | ps3_osrel | 
 public | stream0   | {ps0,ps1,ps2}
 public | stream1   | {ps3}
(6 rows)

CREATE VIEW ps4 AS SELECT id::text FROM stream0;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |   name    |      queries      
--------+-----------+-------------------
 public | ps0_osrel | 
 public | ps1_osrel | 
 public | ps2_osrel | 
 public | ps3_osrel | 
 public | ps4_osrel | 
 public | stream0   | {ps0,ps1,ps2,ps4}
 public | stream1   | {ps3}
(7 rows)

CREATE FOREIGN TABLE stream2 (x integer) SERVER pipelinedb;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |   name    |      queries      
--------+-----------+-------------------
 public | ps0_osrel | 
 public | ps1_osrel | 
 public | ps2_osrel | 
 public | ps3_osrel | 
 public | ps4_osrel | 
 public | stream0   | {ps0,ps1,ps2,ps4}
 public | stream1   | {ps3}
 public | stream2   | 
(8 rows)

CREATE VIEW ps5 AS SELECT x FROM stream2;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |   name    |      queries      
--------+-----------+-------------------
 public | ps0_osrel | 
 public | ps1_osrel | 
 public | ps2_osrel | 
 public | ps3_osrel | 
 public | ps4_osrel | 
 public | ps5_osrel | 
 public | stream0   | {ps0,ps1,ps2,ps4}
 public | stream1   | {ps3}
 public | stream2   | {ps5}
(9 rows)

DROP VIEW ps0;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |   name    |    queries    
--------+-----------+---------------
 public | ps1_osrel | 
 public | ps2_osrel | 
 public | ps3_osrel | 
 public | ps4_osrel | 
 public | ps5_osrel | 
 public | stream0   | {ps1,ps2,ps4}
 public | stream1   | {ps3}
 public | stream2   | {ps5}
(8 rows)

DROP VIEW ps1;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |   name    |  queries  
--------+-----------+-----------
 public | ps2_osrel | 
 public | ps3_osrel | 
 public | ps4_osrel | 
 public | ps5_osrel | 
 public | stream0   | {ps2,ps4}
 public | stream1   | {ps3}
 public | stream2   | {ps5}
(7 rows)

DROP VIEW ps2;
DROP VIEW ps3;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |   name    | queries 
--------+-----------+---------
 public | ps4_osrel | 
 public | ps5_osrel | 
 public | stream0   | {ps4}
 public | stream1   | 
 public | stream2   | {ps5}
(5 rows)

DROP VIEW ps5;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |   name    | queries 
--------+-----------+---------
 public | ps4_osrel | 
 public | stream0   | {ps4}
 public | stream1   | 
 public | stream2   | 
(4 rows)

DROP FOREIGN TABLE stream2;
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema |   name    | queries 
--------+-----------+---------
 public | ps4_osrel | 
 public | stream0   | {ps4}
 public | stream1   | 
(3 rows)

DROP FOREIGN TABLE stream1;
DROP FOREIGN TABLE stream0 CASCADE;
NOTICE:  drop cascades to view ps4
SELECT schema, name, queries FROM pipelinedb.get_streams() ORDER BY name;
 schema | name | queries 
--------+------+---------
(0 rows)

